### PR TITLE
Copying calldata directly to memory.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/CompilerUtils.cpp [hunk 1]
```diff
@@ -76,35 +76,44 @@ void CompilerUtils::storeInMemoryDynamic(Type const& _type, bool _padToWordBound
...
+		if (type.getLocation() == ByteArrayType::Location::CallData)
+		{
+			m_context << eth::Instruction::CALLDATASIZE << u256(0) << eth::Instruction::DUP3
+					  << eth::Instruction::CALLDATACOPY
+					  << eth::Instruction::CALLDATASIZE << eth::Instruction::ADD;
+		}
+		else
+		{
+			solAssert(type.getLocation() == ByteArrayType::Location::Storage, "Memory byte arrays not yet implemented.");
+			m_context << eth::Instruction::DUP1 << eth::Instruction::SLOAD;
```
Comment (chriseth) [inline libsolidity/CompilerUtils.cpp]: Only changed indentation starting from here.

**Comment by caktux** [comment]
The OSX buildslave got stuck for over 8 hours running strict tests, anything in there that could have caused that?

**Comment by gavofyork** [comment]
networking tests, perhaps?

**Comment by caktux** [comment]
yup, there's another build stuck after merging #1044

**Pull request merged**